### PR TITLE
Added support for queuing login requests from incoming logins

### PR DESCRIPTION
**Pull request by ethanmoffat**
- All password-related operations are invoked on a background thread (newly introduced AsyncOperation)
- Incoming login requests return a SERVER_BUSY response if too many concurrent requests are being processed (configurable)
- Add (almost) full test coverage for the Login.cpp handler (missing: seose compatibility test)
- Other minor fixes included along the way

### CODE REVIEW

**Comment by iandinwoodie** [comment]
I'll review real quick. Can you confirm that vscode allows (i.e., ignores) JSON comments in its configuration files?

**Comment by ethanmoffat** [comment]
> I'll review real quick. Can you confirm that vscode allows (i.e., ignores) JSON comments in its configuration files?

Yeah the comments in the config files have been working fine for me. vscode my dev environment so everything I check in works well with it

File: src/world.hpp [hunk 1]
```diff
@@ -124,7 +126,9 @@ class World
 
 		int admin_count;
 
-		World(std::array<std::string, 6> dbinfo, const Config &eoserv_config, const Config &admin_config);
+		World(const std::shared_ptr<DatabaseFactory>& databaseFactory, const Config &eoserv_config, const Config &admin_config);
```
Comment (iandinwoodie) [inline src/world.hpp:129]: Any reason databaseFactory is being passed as a ref to a `shared_ptr` rather than just passing the `shared_ptr` itself?

**Review by iandinwoodie — APPROVED** [review]
Very cursory review, but I've been following the commits recently and adding comments when I felt I had a comment worth sharing.

File: src/world.hpp (hunk 1 above)
Comment (ethanmoffat) [reply src/world.hpp:129]: Nah. I can change it. I think I was trying to use unique_ptr at first but it made it hard to test so I went with shared_ptr instead

**Pull request merged**